packages/browser/src/node/providers/playwright.ts

```diff
@@ -85,11 +95,15 @@ export class PlaywrightBrowserProvider implements BrowserProvider {
     }
 
     const browser = await this.openBrowser()
-    const context = await browser.newContext({
+    const options = {
       ...this.options?.context,
       ignoreHTTPSErrors: true,
       serviceWorkers: 'allow',
-    })
+    } satisfies BrowserContextOptions
+    if (this.ctx.config.browser.launchViewport.maximized) {
```

Comment:
What I _was_ proposing is using the `browser.viewport` values for non-ui instances

Reply:
we cannot use that entry, that viewport is about the iframe not the Vitest UI (top context)

Reply:
how can we display Vitest UI in mobile viewport? no makes sense

Reply:
> for non-ui instances

When UI is disabled

Reply:
when ui is disabled this PR doesn't modify the options, and so the options are passed to the provider (I'll check it again, PW for sure in the `newContext` call => `createContext` in playwright.ts module)

Reply:
we need to review the logic to ensure the iframe is "visible", if we use the viewport size in headless mode, the iframe maybe not being "displayed" properly

Reply:
sorry Vlad, forgot it...

Reply:
adding viewport size when ui disabled...

Reply:
too many things :sweat_smile:

Reply:
done, included also a new `viewport.test.ts`

Reply:
tested with WDIO with chrome, edge and firefox with and without headless in my local, cannot test safari...